File: The Descent.py
import sys
import math

# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
while True:
    Targets = []
    Targets_Sorted = []
    for i in range(8):
        Targets.append(int(input()))  # represents the height of one mountain.
    print("Inputs: " + str(Targets), file=sys.stderr)
    # Write an action using print
    # To debug: print("Debug messages...", file=sys.stderr)
    print("List Targets: " + str(Targets), file=sys.stderr)
    Targets_Sorted = Targets[:]
    print("List Targets Sorted: " + str(Targets_Sorted), file=sys.stderr)
    Targets_Sorted.sort(reverse=True)
    print("Sorted list: " + str(Targets_Sorted), file=sys.stderr)
    
    #for value in Targets_Sorted:
    print(Targets.index(Targets_Sorted[0]))
   
    # The index of the mountain to fire on.
    #print("4")

File: The Descent.java
import java.util.*;
import java.io.*;
import java.math.*;

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/

 
class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        
        ArrayList<Integer> mountain = new ArrayList<Integer>();
        ArrayList<Integer> mountainSorted = new ArrayList<Integer>();

        // game loop
        while (true) {
            for (int i = 0; i < 8; i++) {
                int mountainH = in.nextInt(); // represents the height of one mountain.
                mountain.add(mountainH);
                //System.err.println(mountain);
                mountainSorted.add(mountainH);
                //System.err.println(mountainSorted);
            }

            // Write an action using System.out.println()
            // To debug: System.err.println("Debug messages...");
            
        Collections.sort(mountainSorted, Collections.reverseOrder());
        System.err.println("Sorted list: " + mountainSorted);
        System.out.println(mountain.indexOf(mountainSorted.get(0)));
        mountain.clear();
        mountainSorted.clear();
            
          
        }
    }
}